frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/users/UserPermissionListModel.java

```diff
@@ -86,6 +92,41 @@ private void setAddRoleToUserCommand(UICommand value) {
         addRoleToUserCommand = value;
     }
 
+    public UICommand getRemoveDirectRolesFromUserCommand() {
+        return removeDirectRolesFromUserCommand;
+    }
+
+    public void setRemoveDirectRolesFromUserCommand(UICommand removeDirectRolesFromUserCommand) {
+        this.removeDirectRolesFromUserCommand = removeDirectRolesFromUserCommand;
+    }
+
+    public void removeAllDirect() {
+        List<Permission> elementsToRemove = getDirectPermissions();
+
+        // Correctly set permissions to delete into the selected items.
+        if (!elementsToRemove.isEmpty()) {
+            setSelectedItem(elementsToRemove.get(0));
+        }
+        setSelectedItems(elementsToRemove);
+
+        remove();
```

Comment:
Can those be inside the above `if`?
If the list is empty, is it worth calling those?

Reply:
I lightly change logic and now do nothing when we have nothing to delete.